### ISSUE TITLE
CA1801 false positive on record with property-targeting attribute on parameter

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/roslyn-analyzers#4397]
### Analyzer

**Diagnostic ID**: [CA1801](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1801): `Review unused parameters`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100-rc.2](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

```csharp
    public sealed record AuthProfileImageUrls(
        [property: JsonPropertyName("px_16x16")]
        Uri PixelSize16,
        [property: JsonPropertyName("px_50x50")]
        Uri PixelSize50,
        [property: JsonPropertyName("px_170x170")]
        Uri PixelSize170);
```

If **any** of the parameters have a `[property:` attribute, then CA1801 will fire on **all** parameters.

## Additional context

I wonder the relationship between CA1801 and IDE0060. IDE0060 used to fire a false positive on this, but get fixed later. Once IDExxxx analyzers are available in CLI builds, I will consider to use IDE0060 to replace CA1801.

**Comment by mavasani** [comment]
> Once IDExxxx analyzers are available in CLI builds, I will consider to use IDE0060 to replace CA1801.

They are available in CLI. Please see https://docs.microsoft.com/dotnet/fundamentals/code-analysis/overview#code-style-analysis

**Comment by billybraga** [comment]
How can we be alerted when this is released ? It's included in roslyn-analyzers 5.0.3, but .NET SDK 5.0.201 (most recent) ships with version 5.0.2 of roslyn-analyzers.

**Comment by jmarolf** [comment]
this is not going to be ported to the 5.0.2xx SDKs but will appear in 5.0.3xx

**Comment by billybraga** [comment]
I'm still having this issue in 5.0.203

**Comment by Sharparam** [comment]
Having this issue on 5.0.400, which is strange considering it was going to be included in 5.0.3xx?

**Comment by dlouwers** [comment]
Also having this on records with default parameters

**Comment by dlouwers** [comment]
Workaround:
```public record MyRecord(string P1 = null, string P2 = null);```

to 

```
public record MyRecord
{
  public string P1 { get; init; }
  public string P2 { get; init; }

  public MyRecord(string p1 = null, string p2 = null)
  {
    P1 = p1;
    P2 = p2;
  }
}
```
Very verbose and hope this false positive will be fixed so think it is better to suppress for now.

**Comment by AlbertoMonteiro** [comment]
@dlouwers I think that using that way is much simplier

```csharp
#pragma warning disable CA1801
public sealed record AuthProfileImageUrls(
    [property: JsonPropertyName("px_16x16")] Uri PixelSize16,
    [property: JsonPropertyName("px_50x50")] Uri PixelSize50,
    [property: JsonPropertyName("px_170x170")] Uri PixelSize170);
    ```

**Comment by chibanemourad** [comment]
I have the same problem (CA1801 false positive) with sdk version 5.0.402 : 
```csharp
public abstract record ATest(int Id, string Name);
```
```csharp
public record ConcreteTest(int Id, string Name, State state) : ATest(Id, Name);
```

![image](https://user-images.githubusercontent.com/1395401/158407274-afd96053-fe31-455b-a1ae-cd3ee6ef3149.png)
